app/models/service_orchestration.rb

```diff
@@ -92,26 +93,46 @@ def add_stack_to_resource
     add_resource!(@orchestration_stack) if @orchestration_stack
   end
 
+  def link_orchestration_template
+    # some orchestration stacks do not have associations with their templates in their provider, we can link them here
+    return if @orchestration_stack.nil? || @orchestration_stack.orchestration_template
+    @orchestration_stack.update_attributes(:orchestration_template => orchestration_template)
+  end
+
   def assign_vms_owner
     all_vms.each do |vm|
       vm.update_attributes(:evm_owner_id => evm_owner_id, :miq_group_id => miq_group_id)
     end
   end
 
   def build_stack_create_options
-    # manager from dialog_options overrides the one copied from service_template
     dialog_options = options[:dialog] || {}
-    manager_from_dialog = OptionConverter.get_manager(dialog_options)
-    self.orchestration_manager = manager_from_dialog if manager_from_dialog
-    raise _("orchestration manager was not set") if orchestration_manager.nil?
-
-    # orchestration template from dialog_options overrides the one copied from service_template
-    template_from_dialog = OptionConverter.get_template(dialog_options)
-    self.orchestration_template = template_from_dialog if template_from_dialog
+    pick_orchestration_manager(dialog_options)
+    pick_orchestration_template(dialog_options)
 
     build_stack_options_from_dialog(dialog_options)
   end
 
+  # The order to pick the orchestration manager and template
+  # 1. The ones directly set through setter
+  # 2. The ones set through dialog options
+  # 3. The ones copied from service_template
+  def pick_orchestration_manager(dialog_options)
+    if orchestration_manager == service_template.orchestration_manager
```

Comment:
@Ladas This condition checks whether the manager has been modified through user script. The manager set by script should be honored first. Else we will check the dialog.